test/chart/FunnelChart.spec.tsx

```diff
@@ -229,4 +229,55 @@ describe('<FunnelChart />', () => {
     const activeShape = container.querySelectorAll('.recharts-active-shape');
     expect(activeShape).toHaveLength(0);
   });
+
+  test('renders customized shape when shape is set to be a function', () => {
```

Comment:
Just a curious idea: 

How about instead of duplicating these tests per component, why not create a single test file, for shape, and test all supported components at once with a parametrised test?

Reply:
That would further enforce standardised behaviour and reduce copy pasted code /test cases

Reply:
I did this for the Funnel cases, can you take a look and confirm I'm on the right path. If so, I'll convert the others.